data/code/lib.c

```diff
@@ -216,3 +217,186 @@ int itemshufflemgr_calcslot_ex(const unsigned int* const itemshufflemgr,
 }
 
 #endif
+
+#if SECTIONED_COURSES
+
+static char passed;
+static unsigned int sectionCount = 0;
+
+// override the lap count in a section course to be the number of section points
+void sectionLapOverride()
+{
+    register unsigned short reg3 asm ("r3");
+    register unsigned short reg9 asm ("r9");
+    asm("or %r0,%r3,%r3 \t\n");
+
+    reg9 = (unsigned short)sectionCount;
+
+    if (reg9 != 0)
+    {
+        if (reg9 > 9)
+        {
+            asm("li %r10, 0x9       \t\n"
+                "sth %r10,0x2e(%r31) \t\n");
+        }
+        else
+        {
+            asm("or %r10, %r9, %r9  \t\n"
+                "sth %r10,0x2e(%r31) \t\n");
+        }
+    }
+}
+
+void resetSectionCount()
+{
+    asm("or %r31, %r3, %r3  \t\n");             // do this first
+    sectionCount = 0;
+}
+
+// count the section points to jury-rig the "max laps" count to always be the section number
+void countSectionPoints()
+{
+    register unsigned int *reg0 asm ("r0");
+    register unsigned int reg3 asm ("r3");
+
+    asm("stw %r4, 0x8(%r31)                 \t\n");             // do this first
+
+    // is the "section point" bit set?
+    asm("lbz   %r9, 0x0018 (%r4)            \t\n" // r9 = *r4 + 0x18 ("shortcut point" flag)
```

Comment:
Question: In the inline comment, it says `r9 = *r4 + 0x18`, but I think this would be `r9 = *(r4 + 0x18)` (note the added brackets).